tekton/pipeline_run.go

```diff
@@ -142,8 +146,8 @@ func (r *ReleasePipelineRun) WithOwner(release *v1alpha1.Release) *ReleasePipeli
 	return r
 }
 
-// WithReleaseAndApplicationLabels adds Release and Application labels to the release PipelineRun.
-func (r *ReleasePipelineRun) WithReleaseAndApplicationLabels(releaseName, releaseNamespace, releaseWorkspace string, applicationName string) *ReleasePipelineRun {
+// WithReleaseAndApplicationMetadata adds Release and Application labels to the release PipelineRun.
+func (r *ReleasePipelineRun) WithReleaseAndApplicationMetadata(releaseName, releaseNamespace, releaseWorkspace, applicationName string, release *v1alpha1.Release, md metadata.MetaCopy) *ReleasePipelineRun {
```

Comment:
If we're going to pass the entire release object in, can we stop passing in the name, namespace, and workspace of the release and just retrieve those values from the release object?

Reply:
Yep. That's the plan :wink: